modules/search/search.go

```diff
@@ -160,26 +164,79 @@ func BuildTemplatedQuery(from int, size int, mustClauses []interface{}, shouldCl
 	return &q
 }
 
-func BuildMustClauses(datasource string, category string, subcategory string, richCategory string, username string, userid string) []interface{} {
-	mustClauses := []interface{}{}
-
-	// Check and add conditions to mustClauses
+func BuildDatasourceClause(datasource string, skipVerifyEnabled bool) interface{} {
+	var datasourceClause interface{}
 	if datasource != "" {
 		if strings.Contains(datasource, ",") {
 			arr := strings.Split(datasource, ",")
-			mustClauses = append(mustClauses, map[string]interface{}{
-				"terms": map[string]interface{}{
-					"source.id": arr,
-				},
-			})
+			if skipVerifyEnabled {
+				datasourceClause = map[string]interface{}{
+					"terms": map[string]interface{}{
+						"source.id": arr,
+					},
+				}
+			} else {
+				enabledSourceIDs, err := common.FilterEnabledDatasourceIDs(arr)
+				if err != nil {
+					panic(err)
+				}
+				if len(enabledSourceIDs) > 0 {
+					datasourceClause = map[string]interface{}{
+						"terms": map[string]interface{}{
+							"source.id": enabledSourceIDs,
+						},
+					}
+				} else {
+					datasourceClause = map[string]interface{}{
+						"match_none": map[string]interface{}{},
```

Comment:
why we need a `match_none` here. if the datasource limit condion is not necessary to exists then it shoulld return nothing.